Empty State Design

dangrous opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->
cc @shawnborton 

### Details
<!-- Explanation of the change or anything fishy that is going on -->
This is updating the new chat imagery to match the new designs. This shows when a new chat is created, or when you scroll to the top of an existing chat.

Some things to note:
* The original designs were not in dark mode, they have been adapted appropriately.
* The new headline font is not available yet; this currently only changes the sizing (but it will be a quick swap in the styles file when it is ready). I believe this should NOT block this PR, however. Is that correct @shawnborton?
* There is one spelling fix I noticed in an unrelated translation name. If we feel strongly I can remove this, but it was a quick fix (and I made sure that it is changed in all usages)
* Translations and copy are currently being reviewed/approved but I wanted to get this out for review sooner rather than later.
* The code handling the background image, frankly, a bit messy. React Native was not playing well with the background image and so it feels a bit hacky. However, after about a week of struggle, this is the best I could come up with. If you can figure out a better alternative please let me know! I will not be sad if you criticize.

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/12256


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
1. Create a new chat with any user and make sure the visuals match the design.
2. Create a new group chat with 2 users and make sure the visuals match the design (this should not be different except for having more than one user icon).
3. Create a new group chat with 6-8 users and make sure the visuals match the design (this should not be different except for having more than one user icon).
4. Create a new room and make sure the visuals match the design (this should not be different other than the text changes).

- [X] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->
1. Create a new chat with any user and make sure the visuals match the design.
2. Create a new group chat with 2 users and make sure the visuals match the design (this should not be different except for having more than one user icon).
3. Create a new group chat with 6-8 users and make sure the visuals match the design (this should not be different except for having more than one user icon).
4. Create a new room and make sure the visuals match the design (this should not be different other than the text changes).

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
3. Upload an image via copy paste
4. Verify a modal appears displaying a preview of that image
--->
1. Create a new chat with any user and make sure the visuals match the design.
2. Create a new group chat with 2 users and make sure the visuals match the design (this should not be different except for having more than one user icon).
3. Create a new group chat with 6-8 users and make sure the visuals match the design (this should not be different except for having more than one user icon).
4. Create a new room and make sure the visuals match the design (this should not be different other than the text changes).

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->
![Screen Shot 2022-12-22 at 16 11 13](https://user-images.githubusercontent.com/5487802/209228367-969cefee-1d17-46f4-b204-2d745ce2fbbb.png)
![Screen Shot 2022-12-22 at 16 11 20](https://user-images.githubusercontent.com/5487802/209228406-31648901-5f8c-4d0b-9e9c-c2065d148d23.png)
![Screen Shot 2022-12-22 at 16 11 31](https://user-images.githubusercontent.com/5487802/209228418-fd9eef45-b32a-4fc9-a76d-df53d38ec3f6.png)
![Screen Shot 2022-12-22 at 16 11 39](https://user-images.githubusercontent.com/5487802/209228430-cd796e4d-cc1d-478e-9943-bb8cd588dd22.png)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->
![Screen Shot 2022-12-22 at 16 15 59](https://user-images.githubusercontent.com/5487802/209228454-e89b921b-4f77-4485-acff-bf3e2a84c734.png)
![Screen Shot 2022-12-22 at 16 16 12](https://user-images.githubusercontent.com/5487802/209228480-b3a34f31-2960-4f53-baf9-f86607a44f27.png)
![Screen Shot 2022-12-22 at 16 16 45](https://user-images.githubusercontent.com/5487802/209228494-b6269aee-f596-4af3-9390-aa9e0e548c5b.png)
![Screen Shot 2022-12-22 at 16 16 55](https://user-images.githubusercontent.com/5487802/209228500-0a36815e-ad29-4dd0-885e-7bf66c9cbfaf.png)

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
![Screen Shot 2022-12-22 at 16 17 30](https://user-images.githubusercontent.com/5487802/209228517-6594f222-3320-48ba-b05b-51fc05433d79.png)
![Screen Shot 2022-12-22 at 16 17 47](https://user-images.githubusercontent.com/5487802/209228521-a71aa049-488a-40d8-8f28-ae72443f9548.png)
![Screen Shot 2022-12-22 at 16 18 05](https://user-images.githubusercontent.com/5487802/209228534-dc138095-da4e-4fa5-b444-b12b0dabdd35.png)
![Screen Shot 2022-12-22 at 16 17 36](https://user-images.githubusercontent.com/5487802/209228542-ca6716fe-7889-47ca-92e2-c1a2e12dd356.png)

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->
![Screen Shot 2022-12-22 at 16 25 56](https://user-images.githubusercontent.com/5487802/209228759-b2d6d536-7e55-4b56-b23d-94021504df26.png)
![Screen Shot 2022-12-22 at 16 08 22](https://user-images.githubusercontent.com/5487802/209228795-46233e90-d058-4d9f-915e-f87dfd7ba5c0.png)
![Screen Shot 2022-12-22 at 16 08 54](https://user-images.githubusercontent.com/5487802/209228824-7794fd4d-5272-4a5d-8c35-9738544c1784.png)
![Screen Shot 2022-12-22 at 16 08 32](https://user-images.githubusercontent.com/5487802/209228849-51160dfb-def0-4032-802f-08386ba3b1ac.png)

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->
![Screen Shot 2022-12-22 at 16 09 38](https://user-images.githubusercontent.com/5487802/209228706-f3cb515c-b31e-4469-a496-49c4654669aa.png)
![Screen Shot 2022-12-22 at 16 09 45](https://user-images.githubusercontent.com/5487802/209228714-54cca159-2ff3-40cc-891c-72a472a48cab.png)
![Screen Shot 2022-12-22 at 16 09 53](https://user-images.githubusercontent.com/5487802/209228721-82b4a46e-82d9-421e-88c7-5b2b3713a3e6.png)
![Screen Shot 2022-12-22 at 16 10 03](https://user-images.githubusercontent.com/5487802/209228727-eb52f14b-7c24-4a55-abf7-8e5369e1c50c.png)

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->
![Screen Shot 2022-12-22 at 16 13 50](https://user-images.githubusercontent.com/5487802/209228632-bc37e0cb-51b0-453d-88a2-224ea42f89fb.png)
![Screen Shot 2022-12-22 at 16 14 09](https://user-images.githubusercontent.com/5487802/209228638-9dea104a-8f01-4371-a535-0d83196b98cf.png)
![Screen Shot 2022-12-22 at 16 14 40](https://user-images.githubusercontent.com/5487802/209228652-fd81d3ed-492a-463b-814c-9c1f560f5482.png)
![Screen Shot 2022-12-22 at 16 14 53](https://user-images.githubusercontent.com/5487802/209228660-215314c7-de43-4633-8347-0be129413e9a.png)

</details>

Commit:
initial text formatting

Commit:
Merge branch 'main' into dangrous-emptystatedesign

Commit:
add svg

Commit:
allow for review

Commit:
add background image to empty state

Commit:
make text styling work

Commit:
clean code, fix unrelated spelling error

Commit:
revert Podfile

Commit:
address review and lint

Commit:
fix padding

dangrous commented:
Updated based on your notes @shawnborton! Just to note, I think there's a bit of extra padding when it's a FULLY new chat, vs. one at the top of an existing chat. This is what the new designs look like with messages below them, which I think is 32px (so matching the new mock). Looks like it's more like 40 when there aren't other messages, which I think is padding for the compose bar maybe?

![Screen Shot 2022-12-20 at 15 50 52](https://user-images.githubusercontent.com/5487802/208764293-0a62fd50-fd08-49ea-826d-1f666983f690.png)

shawnborton commented:
Ah got it, thanks for sharing. Well, maybe we should just standardize on `padding: 40px` for all sides (vertical, horizontal) and all scenarios (totally new chat, or existing chat). I think that will make things easier for us, thoughts?

Commit:
update spanish translations

Commit:
update english

Commit:
merge main

dangrous commented:
So the trick with the bottom padding is that this "action item" - i.e. the welcome message - has its own padding, and then the full list of items (including any chat messages) also has ITS own padding. When there are other messages, we only see the padding of this welcome message (as the list padding is below the messages). When there aren't we see both together, which is why it's bigger when there aren't messages. I can't see an easy way to make it the same in both situations... Basically it'll always be ~8px more (it's actually more, but the chat messages have their own padding so that's what it looks like). Should we keep it as is in this PR for now? The difference isn't huge.

In other news, translations and copy are ready to go so this is good to test/review!

shawnborton commented:
Cool, I think it's fine to keep as-is then, thanks for explaining!

shawnborton commented:
For the mobile sizes, I wonder if we can make the background a little bit smaller so we squeeze in some more detail from the illustration? Something more like this: 

![image](https://user-images.githubusercontent.com/2319350/208972637-754891b7-0535-4692-99a9-70e79ea71a1e.png)

Commit:
update image sizing

dangrous commented:
Updated based on @shawnborton's feedback (screenshots updated as well). Man, every time I think "this will be an easy fix" the image conspires to make things more challenging. This is messy but still works, I think. As always, clean up ideas are appreciated.

Commit:
add trailing comma

grgia commented:
Is there a specific part you think could be cleaned up?

dangrous commented:
@grgia great question! Sorry I hadn't clarified. Basically it's this section which I hatteeeeee:

https://github.com/Expensify/App/pull/13647/files#diff-03a581d113c83bcb1ecd2620607374359631282d68421d9b0eccd85db6f72c20R44-R49 (the weird handling of the height of the image)

And there was another section but I just realized I commented it out and never deleted it. So, I'm going to do that now.

Commit:
remove unnecessary view

shawnborton commented:
Nice, this is looking good! Can you share screenshots for chats with multiple people (a group of 2, then a group of 6-8), as well as new rooms?

dangrous commented:
Coming in hot!

<details>
<summary>Chat with 2 others</summary>

Web
![Screen Shot 2022-12-22 at 13 33 27](https://user-images.githubusercontent.com/5487802/209204616-d0a47c3c-e801-432f-b221-e9c9019d4ae6.png)
Mobile Chrome
![Screen Shot 2022-12-22 at 13 39 12](https://user-images.githubusercontent.com/5487802/209204633-60add314-097e-4bf2-9e75-37ef944c61a0.png)
Mobile Safari
![Screen Shot 2022-12-22 at 13 34 49](https://user-images.githubusercontent.com/5487802/209204638-8355f9bf-f52d-44f4-b0f4-61479029d918.png)
Desktop
![Screen Shot 2022-12-22 at 13 30 09](https://user-images.githubusercontent.com/5487802/209204650-e702cffb-eb48-46da-afe2-12f8c2dc3f8b.png)
Android
![Screen Shot 2022-12-22 at 13 28 41](https://user-images.githubusercontent.com/5487802/209204661-fd32424a-91ea-42b4-a97e-2c41bcdabd8c.png)
iOS
![Screen Shot 2022-12-22 at 13 18 10](https://user-images.githubusercontent.com/5487802/209204672-256e0d07-bb03-4340-b900-17c086a65de6.png)
</details>

<details>
<summary>Chat with 7 others</summary>

Web
![Screen Shot 2022-12-22 at 13 33 39](https://user-images.githubusercontent.com/5487802/209204684-534c9284-a38f-4fda-b612-c42e0fb6f0c4.png)
Mobile Chrome
![Screen Shot 2022-12-22 at 13 39 37](https://user-images.githubusercontent.com/5487802/209204690-ffb8be50-033a-49d6-9e6b-fea9f54da001.png)
Mobile Safari
![Screen Shot 2022-12-22 at 13 35 07](https://user-images.githubusercontent.com/5487802/209204695-ae01a18d-8742-434e-9b86-5245d2539f95.png)
Desktop![Screen Shot 2022-12-22 at 13 30 42](https://user-images.githubusercontent.com/5487802/209204702-ea212ed0-a7f1-4500-aedb-88b436d7fbcc.png)
Android
![Screen Shot 2022-12-22 at 13 29 03](https://user-images.githubusercontent.com/5487802/209204717-aa654c57-52db-45ef-b25c-b755dd072169.png)
iOS
![Screen Shot 2022-12-22 at 13 19 03](https://user-images.githubusercontent.com/5487802/209204727-caa9681b-3b21-4d8e-aa8b-5587b0c6e4e3.png)
</details>

<details>
<summary>Room</summary>

Web
![Screen Shot 2022-12-22 at 13 33 19](https://user-images.githubusercontent.com/5487802/209204737-b7f2f516-d9ca-45cd-ad3a-4eb921cfe401.png)
Mobile Chrome
![Screen Shot 2022-12-22 at 13 38 53](https://user-images.githubusercontent.com/5487802/209204747-9f5029c3-d605-4b95-93d1-5b2424f5f1a8.png)
Mobile Safari
![Screen Shot 2022-12-22 at 13 34 36](https://user-images.githubusercontent.com/5487802/209204755-9b16cdd8-c117-48c7-9069-a74b6edc3156.png)
Desktop
![Screen Shot 2022-12-22 at 13 32 45](https://user-images.githubusercontent.com/5487802/209204770-7dab1388-8245-403f-90be-c34233d7e96c.png)
Android
![Screen Shot 2022-12-22 at 13 28 11](https://user-images.githubusercontent.com/5487802/209204778-cb9d9752-696d-440e-b99e-b273085ddc3b.png)
iOS
![Screen Shot 2022-12-22 at 13 19 21](https://user-images.githubusercontent.com/5487802/209204785-b498697e-4fba-447e-b935-92243418a91d.png)
</details>

shawnborton commented:
Okay sweet, thanks for sharing! It looks like for the group chats with multiple people, the avatar is higher up against the background than say a one person chat?

One | Mutliple
----- | -----
<img width="409" alt="image" src="https://user-images.githubusercontent.com/2319350/209206975-f6985bf8-6f90-4839-8e30-e2230fca7dd2.png"> | <img width="416" alt="image" src="https://user-images.githubusercontent.com/2319350/209206876-709b22fe-4796-4391-ab07-5f0020391c79.png">

dangrous commented:
hm yeah I think it is a bit higher - the background image is based on the height of that whole block, so when there are more names  (and therefore that text block is bigger) it looks like it's higher up.

shawnborton commented:
Ah okay, thanks for clarifying. Anything we can do to fix that so we get a consistent experience across various chat types?

dangrous commented:
What about this? It's a little hard to confirm because the icons are different sizes but it's roughly a little less than halfway down the avatars on all three

![image](https://user-images.githubusercontent.com/5487802/209224508-c4454bfc-e370-4cdb-9cd5-87276a0f33aa.png)

shawnborton commented:
Nice! This feels great to me, thanks for the adjustments Daniel!

dangrous commented:
Great! I'll get the code up and the screenshots updated.

Commit:
update spacing to be uniform across chat types

dangrous commented:
Updated!

grgia commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -38,28 +41,46 @@ const defaultProps = {
+
+    // The image handling depends on how large the chat window is
+    const backgroundImageHeight = Dimensions.get('window').height < Dimensions.get('window').width
+        ? 450
+        : 300;
```
What's the reasoning behind 450 / 300?

dangrous replied:
The 300 is what was needed to match the iOS mocks. The 450 is the smallest possible height to have it not cut off by the container top edge on desktop at bigger screen sizes. I wish I had better answers for you, haha.

grgia replied:
in that case, maybe we should add these to the `variables.js` file and reference from there? Then you could use a more descriptive name like imagePortraitWidth / imageLandscapeWidth (just an example)

grgia replied:
Follow up: Is it possible to use % here so you wouldn't have to reset the height and width later?

dangrous replied:
I think I could swap to percentage here but it would still have to be two different values... I can look into it though!

grgia commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -38,28 +41,46 @@ const defaultProps = {
     const icons = ReportUtils.getIcons(props.report, props.personalDetails, props.policies);
+
+    // The image handling depends on how large the chat window is
+    const backgroundImageHeight = Dimensions.get('window').height < Dimensions.get('window').width
```
```suggestion
    const backgroundImageHeight = (Dimensions.get('window').height < Dimensions.get('window').width)
```
NAB, I think this is more readable

grgia commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -38,28 +41,46 @@ const defaultProps = {
             onClose={() => Report.navigateToConciergeChatAndDeleteReport(props.report.reportID)}
         >
+            <View style={[styles.emptyStateBackgroundContainer, {
+                // since height and top are dynamically set they needs to be updated here, not in styles.js
+                height: backgroundImageHeight,
+                top: -backgroundImageHeight,
+            }]}
+            >
```
When you say "dynamically set" do you mean as the user changes window size?

dangrous replied:
I guess `dynamically` is wrong. It's only set when the report first loads. If there's an easier way to tell between landscape and portrait I'm all ears.

grgia replied:
Okay in terms of cleanup- one idea is moving all of this (including the backgroundImageHeight variable declaration) into StyleUtils and then calling StyleUtils.get_____() to return the correct style object

dangrous replied:
oh neat, I didn't know `StyleUtils` existed! That's a great idea

grgia commented on src/components/ReportWelcomeText.js:
```diff
@@ -62,66 +63,74 @@ const ReportWelcomeText = (props) => {
-        <Text style={[styles.mt3, styles.mw100, styles.textAlignCenter]}>
-            {isPolicyExpenseChat && (
-                <>
-                    {/* Add align center style individually because of limited style inheritance in React Native https://reactnative.dev/docs/text#limited-style-inheritance */}
-                    <Text style={styles.textAlignCenter}>
```
Is it okay to remove this?

dangrous replied:
Yep! Removing that style shifts the text over to the left, to match the mocks. Not needed anymore.

grgia commented on src/styles/styles.js:
```diff
@@ -2860,6 +2860,21 @@ const styles = {
         maxWidth: 240,
     },
+
+    welcomeTextHeader: {
+        fontSize: variables.fontSizeHero,
+    },
```
perhaps this should just be `heroFont` or something and placed near the font styles so we could reuse this style?

dangrous replied:
That works for me! Maybe something like `headerFont` since I think we'll put the new font in here as well once it exists

grgia commented on src/styles/styles.js:
```diff
@@ -2860,6 +2860,21 @@ const styles = {
+    emptyStateBackgroundImage: {
+        width: '100%',
+        height: '80%',
+        bottom: '-40%',
```
Real quick, what's the reasoning behind these %?

dangrous replied:
Sorta same as above. It's what made it work, haha. Height and width need to be defined or else the image doesnt' show up. The width needs to be 100% or greater or else you see the edges on desktop, and the height was just the value that made it the right size for the mocks. The `bottom` moves it up to the right level to match.

dangrous replied:
Oh interesting, I didn't know that about FlatList. I didn't notice any visible issues when scrolling through longer chats, but would be good to test more. As for the percentages, it was basically just the values that made things work. I can try to find pixel values instead, and then can add them to the CONST file.

dangrous replied:
Update on this - I use the percentages here vs. pixel sizes since the size of the container changes depending on whether it's wide screen or not. Pixel sizes would make it go weird on one or the other.

grgia reviewed (commented):
Left a couple of comments that came up while looking over this!

Commit:
address comments

dangrous commented:
Updated! I also found out there's an existing variable for whether or not it's portrait or landscape so I brought that in with a prop. This should be much cleaner.

shawnborton commented:
Nice, this is looking pretty good! A couple of things:

I wonder if we want all of the empty state avatar sizes to be the same larger size that we use for a single person/room chat? Like so:
![image](https://user-images.githubusercontent.com/2319350/209956413-e712072b-2b48-4c71-97e6-d5a17a70d612.png)


Another thing - could you show us a screenshot where the screen is really really wide? Just want to take a look at some edge cases here, thanks!

dangrous commented:
@shawnborton Yeah that makes sense about the icons, let me see what I can do.

And yep! Will add a very wide screen screenshot as well. I *think* it should be okay for all *reasonable* screensizes, the top of the image *will* start to get cut off as the window gets wider but the existing %s, #s, etc. should work with, say, a full screen 1080p monitor (since that's what I tested on haha). Not sure about wider. In any case, will do my best!

parasharrajat reviewed (changes requested):
BUG: WEB: Clicking the empty area adjacent to the avatar opens the report details. 

BUG: Web: No copy image menu should be shown to the user while right-clicking the image. It is a background image and thus should behave like it. 
(You can fix it with `pointerEvents:none`)

https://user-images.githubusercontent.com/24370807/209984071-153c7a74-9cc7-423f-98ca-89075b9c0a37.mp4

Commit:
Merge branch 'main' into dangrous-emptystatedesign

Commit:
update clickability

dangrous commented:
@parasharrajat good catches on the background image clicking and the report details issue! I've fixed those in the latest code.

@shawnborton I think it might be easier to update the avatar sizes in a separate PR, only because the same code is also used for the `ReportDetailsPage` and I think it might get a bit messy to wrap it into this one. Happy to make that PR soon though!

And, here's a screenshot of a too-wide screen. Larger than about 1715 pixels wide will start to have the image cut off at the top (and actually at the bottom as well, though it's harder to see). Is that enough of a good area, or do we need to take another tack? Right now, if we modify it to NOT be cut off at those higher widths, it'll be smaller/shorter overall. I haven't yet figured out a way around this.

<img width="2070" alt="Screen Shot 2022-12-29 at 17 12 18" src="https://user-images.githubusercontent.com/5487802/210016144-cac9183a-280f-4b65-81ed-09c6db9457b2.png">

Lastly, I think @parasharrajat is right about the image looking a little cut off even when it's fully shown. Is that expected?

shawnborton commented:
> I looked into it and I think it might be easier to update the avatar sizes in a separate PR, only because the same code is also used for the ReportDetailsPage and I think it might get a bit messy to wrap it into this one. Happy to make that PR soon though!

Sounds good!

> Right now, if we modify it to NOT be cut off at those higher widths, it'll be smaller/shorter overall. 

I was thinking that we would give this background area some kind of min-height, and the repeating background asset should be as tall as the height of this area. This way it doesn't get cut off, it just repeats infinitely on the x axis. Let me know if that makes sense!

shawnborton commented:
Hmm yeah, I suppose we could fix those. Here is an updated version:
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/2319350/210089532-be0af8cb-b62a-48ea-9693-42ad826bc334.png">


[empty-state_background-fade.png.zip](https://github.com/Expensify/App/files/10325431/empty-state_background-fade.png.zip)

@dangrous this should be the same size as the previous image, so you can simply overwrite it.

shawnborton commented:
I think it's fine that we use the same asset across both mobile and desktop - we are purposely cropping it a bit different so that we expose some more detail on mobile.

dangrous commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -38,28 +43,36 @@ const defaultProps = {
             onClose={() => Report.navigateToConciergeChatAndDeleteReport(props.report.reportID)}
         >
+            <View pointerEvents="none" style={[styles.emptyStateBackgroundContainer, StyleUtils.getBackgroundImageHeightStyle(props.isSmallScreenWidth)]}>
+                <Image
```
Yeah I went that direction initially, but because the content itself doesn't extend up as high as the background needed to, I was finding it close to impossible to get it to look right and not get cut off at the top. I'll try a little more today, but I'm not hopeful (trust me, I would definitely prefer to use `ImageBackground` if it would work)

dangrous commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -38,28 +43,36 @@ const defaultProps = {
                 ]}
             >
-                <View style={[styles.justifyContentCenter, styles.alignItemsCenter, styles.flex1]}>
+                <View style={[styles.ph5, styles.pb3, styles.alignSelfStart]}>
                     <Pressable onPress={() => ReportUtils.navigateToDetailsPage(props.report)}>
                         <RoomHeaderAvatars
                             icons={icons}
                             shouldShowLargeAvatars={isPolicyExpenseChat}
                         />
                     </Pressable>
+                </View>
```
Let me double check this works but yeah this is cleaner.

dangrous replied:
Works!

dangrous commented:
@shawnborton 

> I was thinking that we would give this background area some kind of min-height, and the repeating background asset should be as tall as the height of this area. This way it doesn't get cut off, it just repeats infinitely on the x axis. Let me know if that makes sense!

So this does make sense, but it was... unexpectedly challenging to actually implement in the app - basically the container that would have the background image isn't big enough and so it was affecting the image in weird ways, cutting it off and scaling wrong, etc.). Since I couldn't figure out a way to use the `ImageBackground` component, I couldn't actually repeat the image on the x axis. I'm going to give it one more try today but (as I mentioned in another comment haha) I'm not hopeful. It would make things nicer/easier, certainly.

Commit:
update background image

Commit:
address comments

Commit:
Merge branch 'main' into dangrous-emptystatedesign

Commit:
merge main, fix conflicts

Commit:
remove unnecessary propType

dangrous commented:
Updated to address all feedback except where otherwise noted. I will keep playing with ImageBackground but I'm not anticipating being able to go this direction, for a number of reasons:
* The image would be lower down and potentially cut off at the top since it's trying to be a background only for that details section and they don't extend as high as we'd like them to. I can add a View that extends the height but it still starts pretty low.
* If we try to repeat the background it repeats vertically not horizontally since that's the shorter dimension. There's a way to fix that in standard HTML but doesn't seem to work in React Native.
Soooo will give it another hour or two but if I don't report back, assume it didn't work... Happy to take ideas as well.

Commit:
add window proptypes

dangrous commented:
Tried ImageBackground for a while, was still not able to make it work. I can't exactly figure out why but whether you use percentages or pixel values, or just leave heights blank, the height of the container does not cooperate with the height of the image, so it's always cut off. I think unfortunately it's not going to work, unless someone with more React Native expertise wants to give it a try

shawnborton commented:
Got it. So what are the most current screenshots you have of the progress so far? Perhaps we're at a close enough point, but at the same time, it can't hurt to solicit some extra eyes in the open source room if you are feeling stuck!

dangrous commented:
Yeah I guess I can try asking in Slack. Here are some examples of the issues:

Good on mobile (but may get weird if the chat is long?), bad on desktop
<img width="1728" alt="Screenshot 2023-01-04 at 10 43 37" src="https://user-images.githubusercontent.com/5487802/210594040-ff27ab99-100a-4e0c-bf64-ba4b1c148534.png">

Not unreasonable on desktop (but too far down), content now not visible on mobile
<img width="1728" alt="Screenshot 2023-01-04 at 10 47 35" src="https://user-images.githubusercontent.com/5487802/210594054-c247641d-cc5d-440e-96fd-5540bca193d3.png">

Commit:
Merge branch 'main' into dangrous-emptystatedesign

Commit:
update header font

dangrous commented:
Hi all! Two things:

* Updated the header font to New Kansas, yay!
* Where do we stand with this PR in terms of review? I know ideally we'd have a slightly different implementation of the background image but it seems like nobody (including myself) is able to adjust it much... Do we want to just push forward with what we have, and maybe take another look further in the future?

grgia commented:
Here's my go using ImageBackground:

https://user-images.githubusercontent.com/38015950/211422703-90286cf9-4e7b-4b8d-93c4-8738c92c5000.mov

<img width="564" alt="image" src="https://user-images.githubusercontent.com/38015950/211422799-9e9705cd-c96e-4629-bc4b-c2c300b09123.png">

What wasn't working?

shawnborton commented:
Oh wow, that is looking great Georgia! Thanks for the assist here.

shawnborton commented:
Also, sorry for the last minute change here but given that we updated our dark mode colors slightly, I think we need to slightly adjust the color of the BG image. Here is a new one (another direct replacement):
![empty-state_background-fade](https://user-images.githubusercontent.com/2319350/211526891-9f82a22f-9a0b-4147-8564-6ba86dcf6a4a.png)

[empty-state_background-fade.png.zip](https://github.com/Expensify/App/files/10381601/empty-state_background-fade.png.zip)

dangrous commented:
@grgia Nice! What's the code? For me I could always get it to look good on one of the platforms (e.g. ios) and then it would start going weird on the other (e.g. desktop). Or resizing the window would mess it up... But I'm sure I was just missing something dumb.

Commit:
update image, change to imagebackground

Commit:
merge conflicts

dangrous commented:
Updated the image, and switched to `ImageBackground`! I think.... we're ready to go?

JmillsExpensify commented:
Agreed, we're so close this is awesome polish. Let's get this one merged!

shawnborton commented:
Woo, can't wait for this one!

Commit:
update error message style

grgia commented:
| Web | Desktop | IOS | Safari | Android | Chrome |
|-----|-----|-----|-----|-----|-----|
|<img width="1609" alt="image" src="https://user-images.githubusercontent.com/38015950/212212195-4e936ce1-92f4-46f4-a6c5-1a854d021307.png">|<img width="1312" alt="image" src="https://user-images.githubusercontent.com/38015950/212212326-a0ceb51e-1c4b-43e1-868d-4b85db6222db.png">|<img width="564" alt="image" src="https://user-images.githubusercontent.com/38015950/212212741-3c964ddf-2aff-40ba-b530-f2164a0872f6.png">|<img width="520" alt="image" src="https://user-images.githubusercontent.com/38015950/212212299-29b8e888-f527-4b12-97bf-4c571b594075.png">|<img width="502" alt="image" src="https://user-images.githubusercontent.com/38015950/212212652-9f06dd81-86d0-4275-ba15-aded43554d8d.png">|<img width="546" alt="image" src="https://user-images.githubusercontent.com/38015950/212212591-15ab7162-5d9b-40d7-b32d-29301106b206.png">


WOO almost there! It looks great on all platforms for an empty chat. But there is a bug for IOS/Android for long chats where you cannot scroll to see the entire image:


https://user-images.githubusercontent.com/38015950/212212978-593a902a-61e4-45d6-8527-56f132d58606.mov

shawnborton commented:
Good catch Georgia! Maybe we can just give the empty state area some min-height or something like that?

Commit:
add minheight to see full image when text in chat

dangrous commented:
Updated! Added a min-height to the full container and pushed everything to the bottom of it, this way it won't mess up any of the formatting for the text/avatars/etc.

grgia commented:
Testing now!

grgia commented:
Quick sanity check:
Should we be allowed to scroll when chat is empty or has only one chat?

https://user-images.githubusercontent.com/38015950/212435338-c820f25b-0bd0-408b-8fa2-114f98465e24.mov

https://user-images.githubusercontent.com/38015950/212435243-d4b4ae71-3191-43d1-9371-3e16a0103087.mov

shawnborton commented:
Good catch - I would think this would behave like it does today where new chats wouldn't have any scroll to them, because the empty state is ideally shorter than the total viewport height. I mean unless the keyboard is up on mobile and the viewport is small, then scrolling might make sense, but in the screenshots you shared above I would think not.

Commit:
fix scrolling issue

dangrous commented:
Okay I think the latest code fixes that scrolling issue. Also cleans things up a little bit

shawnborton commented:
Awesome! @grgia @parasharrajat can you take a look and hopefully we can merge soon? Thanks!

Commit:
merge and add docs

dangrous commented:
Updated to address merge conflicts - would love to get this reviewed and merged this week!

shawnborton commented:
Looks like we have a conflict now @dangrous 

@parasharrajat are you able to finish review on this? Let's try to get this one merged by EOW if possible.

dangrous commented:
Oh ha I think it's the other code I wrote. Looking into it.

Commit:
merge conflicts

dangrous commented:
Updated to fix conflicts!

shawnborton commented:
Awesome, @parasharrajat are you able to take this one over the finish line please? Thank you!

JmillsExpensify commented:
@parasharrajat checking in on this one? It's getting close to two days and we really need to get this merged. Thank you!

grgia commented:
If you're swamped I can do the checklist, just let me know @parasharrajat !

JmillsExpensify commented:
Thanks @parasharrajat!

shawnborton commented:
@parasharrajat can you please provide us with an update on where you are at with this one? You said you would complete it in 2 hours a couple of days ago but I haven't seen any progress since then. If you aren't able to get to this today, then I think we should have Georgia take care of the checklist.

parasharrajat reviewed (changes requested):
UI is looking fine to me. Only a few changes.

shawnborton commented:
> Is this fine by design the "Error for chat and workspace creation failure"?

I think so, because that's how it currently works today right? Or was this something introduced in this PR?

shawnborton commented:
> There are no more tooltips on the displayName on the title summary. Also, clicking the Avatar/displayName does not open the Details page. Is this Correct?

Ah interesting, good catch. Yeah, I would think we would want it to behave like it does today, where clicking on the user name in the empty state space launches the details view. 

In terms of design, I checked the screenshots and this looks good to me. I believe the linked issue has the screenshots we are going off of.

shawnborton commented:
> It used to be center aligned and limited with width. (Staging)

Ah yeah, we decided to make it full width/left align it, so we're all good there.

dangrous commented on src/pages/home/report/ReportActionItemCreated.js:
```diff
@@ -37,28 +43,36 @@ const defaultProps = {
+                    />
+                </View>
+                <View
+                    accessibilityLabel="Chat welcome message"
```
See my comments in Slack - I agree that this should be changed BUT I think we need to handle that separately (for context, this was already here). It is being used for testing [here](https://github.com/Expensify/App/blob/87d197a1ace70c3ca69cfc197d844ec1449a722c/tests/ui/UnreadIndicatorsTest.js#L236-L237) and this is the pattern across the app. Again, I don't like this either but I think we can do another issue for this at this time.

Commit:
move pointerEvent none

dangrous commented:
Updated with the bug fixed about pointer events! I don't think we should change the accessibilityLabel in this PR because it affects other things, but do agree that it should be changed. We can make another issue to investigate things overall, once this is pushed.

shawnborton commented:
Awesome, thanks @dangrous - hopefully we can get this merged soon!

shawnborton commented:
Yes - I think both the click area and the gap between paragraphs is fine.

On Mon, Jan 30, 2023 at 6:36 PM Rajat Parashar ***@***.***>
wrote:

> Should there be a gap between the two lines like the normal chat for room
> chat as well based on the following image?
> [image: image]
> <https://user-images.githubusercontent.com/24370807/215551406-d46fdf2f-1d3f-43fc-8b1d-6700cdac3e32.png>
>
> Group Chat/Single Chat/Normal Chat
>
> [image: Screenshot 2023-01-30 23:05:02]
> <https://user-images.githubusercontent.com/24370807/215551569-fe00a2ae-36d6-4f60-9b72-ca7a6cf7c49f.png>
>
> Room Chat
> [image: Screenshot 2023-01-30 23:06:00]
> <https://user-images.githubusercontent.com/24370807/215551736-2f4693f0-6c9e-4ef9-83e1-a9c94c303ae6.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Expensify/App/pull/13647#issuecomment-1409041026>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARWH5WGPF3MLGUOB6WA2VLWU73ZTANCNFSM6AAAAAATAIECTI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

shawnborton commented:
Looks great! Merge is all you @grgia

parasharrajat reviewed (approved):
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

cc: @grgia 

:ribbon: :eyes: :ribbon:  C+ reviewed

shawnborton commented:
Awesome. @grgia all you for the merge!

JmillsExpensify commented:
Let's click the button on this one! It's a great improvement worth getting in the hands of users.

grgia reviewed (approved):
Looks great!

grgia commented:
Great job on this @dangrous !

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

dangrous commented:
Amazing, thanks all for pushing this through! It's been a long road haha but it looks great.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4069698718) to staging by https://github.com/grgia in version: 1.2.64-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4088843261) to production by https://github.com/thienlnam in version: 1.2.64-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4088843261) to production by https://github.com/thienlnam in version: 1.2.64-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4088843261) to production by https://github.com/thienlnam in version: 1.2.64-7 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|failure ❌ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
